src/Command/FindAndUploadFilesCommand.php

```diff
@@ -177,18 +201,43 @@ protected function execute(InputInterface $input, OutputInterface $output)
         $io->section(
             "Uploading dependency files to Debricked, starting from {$searchDirectory}, ignoring \"{$directoriesToExcludeString}\""
         );
+
+        $repository = \strval($input->getArgument(self::ARGUMENT_REPOSITORY_NAME));
+        $commit = \strval($input->getArgument(self::ARGUMENT_COMMIT_NAME));
+        $zippedRepositoryName = "{$repository}_{$commit}.zip";
+        $zip = new ZipArchive();
+        $zip->open($zippedRepositoryName, ZipArchive::CREATE);
+
         $uploadId = null;
         $uploadedFilePaths = [];
         $progressBar = new ProgressBar($output);
         $progressBar->start();
         $progressBar->setFormat(' %current% file(s) found [%bar%] %percent:3s%% %elapsed:6s% %memory:6s%');
         $this->setProgressBarStyle($progressBar);
+        $uploadAllFiles = \boolval($input->getOption(self::OPTION_UPLOAD_ALL_FILES));
         foreach ($finder as $file) {
-            if (\array_key_exists($fileName = $file->getFilename(), $dependencyFileNames) === true) {
+            $pathName = $file->getPathname();
+            $extension = $file->getExtension();
+            $fileName = $file->getFilename();
+            if (\array_key_exists($extension, $this->blacklist) === false && $uploadAllFiles === true) {
+                $pathArray = explode('/', $pathName);
+                unset($pathArray[1]);
+                $pathNameWithoutSearchDir = implode('/', $pathArray);
+                $zip->addFile($pathName, $pathNameWithoutSearchDir);
+            }
+
+            if (\array_key_exists($fileName, $dependencyFileNames) === true) {
+                if (\array_key_exists($fileName, $dependencyFileNamesRequiresAllFiles) === true && empty($uploadAllFiles) === true) {
+                    $io->warning("Skipping {$pathName}");
+                    $io->warning('Found files which requires that all files needs to be uploaded.');
```

Comment:
`"Skipping {$pathName}.\n\nFound file which requires that all files needs to be uploaded. Please enable the {self::OPTION_UPLOAD_ALL_FILES} option if you want to scan this file."`